### PR TITLE
Complete the mini exercise

### PR DESCRIPTION
**Pull request by DuongPham-dev**
- #48

### CODE REVIEW

File: practices/practice-one/src/scripts/controllers/todo.controller.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        return this.todos;
+    }
+
+    handleAddTodo(text: string) {
+        if (Boolean(text)) {
+            const date = new Date();
+            const currentDate = `${date.getDate()}-${date.getMonth() + 1}-${date.getFullYear()}`
+            const data: TodoType = {
+                id: this.todos.length,
+                description: text,
```
Comment (cunguyen-agilityio) [inline practices/practice-one/src/scripts/controllers/todo.controller.ts]: Why not title?

Comment (phamtanduongtk29) [reply practices/practice-one/src/scripts/controllers/todo.controller.ts:42]: I thought it was the content so I put it like that

Comment (DuongPham-dev) [reply practices/practice-one/src/scripts/controllers/todo.controller.ts]: - I will update it soon

File: practices/practice-one/src/scripts/constants/hash.constant.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+enum TodoAction {
+    ALL = '#/',
+    ACTIVE = '#/active',
+    COMPLETED = '#/completed',
+    DEFAULT = '/'
+}
+
```
Comment (cunguyen-agilityio) [inline practices/practice-one/src/scripts/constants/hash.constant.ts:7]: spaces

File: practices/practice-one/src/scripts/constants/hash.constant.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+enum TodoAction {
```
Comment (cunguyen-agilityio) [inline practices/practice-one/src/scripts/constants/hash.constant.ts]: TodoStates

File: practices/practice-one/src/scripts/controllers/todo.controller.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+            case TodoAction.ACTIVE: {
+                return this.todos.filter((todo) => !todo.isCompleted);
+            }
+            default: {
+            }
+        }
+        return this.todos;
+    }
+
+    handleAddTodo(text: string) {
```
Comment (cunguyen-agilityio) [inline practices/practice-one/src/scripts/controllers/todo.controller.ts]: text: Naming not good

File: practices/practice-one/src/scripts/controllers/todo.controller.ts [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        if (Boolean(text)) {
+            const date = new Date();
+            const currentDate = `${date.getDate()}-${date.getMonth() + 1}-${date.getFullYear()} ${date.getHours()}:${date.getMinutes()}:${date.getSeconds()}`
```
Comment (cunguyen-agilityio) [inline practices/practice-one/src/scripts/controllers/todo.controller.ts]: Can we make this as a utils?

File: practices/practice-one/src/scripts/views/todo-form.view.ts [hunk 6]
```diff
@@ -1,91 +1,208 @@
...
-                    isCompleted: false
-                }
-                this.handleAddTodo(data)
-                this.textElment.value = ''
-                todosElement.classList.add('not-empty')
-            }
+            const text = this.textElment.value.trim();
+            this.todoController.handleAddTodo(text)
+        });
+        this.checkAllElement.addEventListener('click', (e) => { });
```
Comment (cunguyendev) [inline practices/practice-one/src/scripts/views/todo-form.view.ts]: Looks like this event is empty for now/do nothing

File: practices/practice-one/src/scripts/views/todo-form.view.ts [hunk 7]
```diff
@@ -1,91 +1,208 @@
...
+    }
+
+    /**
+   * - Add a new job to do
+   * @param data
+   */
+    handleAddTodo(data: Param) {
+        const todoItem = todoItemTemplate(data);
+        if (this.hash !== '#/completed') {
```
Comment (cunguyendev) [inline practices/practice-one/src/scripts/views/todo-form.view.ts]: Can we  use this as a constant `#/completed`

File: practices/practice-one/src/scripts/views/todo-form.view.ts [hunk 8]
```diff
@@ -1,91 +1,208 @@
...
 
-    private handleCompletedTodo(element) { }
+    /**
+     * - Select a completed to-do
+     * @param element
+     */
+    handleCompletedTodo(element: HTMLLIElement) {
+        const data = element.getAttribute('data-item');
+        this.todoController.handleCompletedTodo(parseInt(data))
+        if (this.hash === '#/completed' || this.hash === '#/active') {
```
Comment (cunguyendev) [inline practices/practice-one/src/scripts/views/todo-form.view.ts]: Also apply constants for this

**Pull request merged**